pkg/util/iptables/iptables.go

```diff
@@ -284,6 +284,42 @@ func (runner *runner) restoreInternal(args []string, data []byte, flush FlushFla
 	return nil
 }
 
+// ParseTableSaveChains parses a table's save and returns a map of Chain to chain in iptables-save format
+// (with counters etc).
+func ParseTableSaveChains(table Table, save []byte) map[Chain]string {
+	// get lines
+	lines := strings.Split(string(save), "\n")
+	chainsMap := make(map[Chain]string)
+	tablePrefix := "*" + string(table)
+	lineNum := 0
+	// find beginning of table
+	for ; lineNum < len(lines); lineNum++ {
+		line := lines[lineNum]
+		if strings.HasPrefix(line, tablePrefix) {
+			lineNum++
+			break
+		}
+	}
+	// parse table lines
+	for ; lineNum < len(lines); lineNum++ {
+		line := lines[lineNum]
+		// end of table
+		if strings.HasPrefix(line, "COMMIT") || strings.HasPrefix(line, "*") {
+			break
+
+			// ignore comments
+		} else if len(line) < 1 || strings.HasPrefix(line, "#") {
```

Comment:
What about leading whitespace?  I guess you should Trim() each line as you read them, which will invoke copies.  SIGH.